### PR TITLE
tilt logs: add port/resource flags, filter by resource [ch8432 ch8433]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @nicks, @jazzdan,

Please review the following commits I made in branch maiamcc/filter-resources:

857bdf022b6d19dc5f413bccb760be26297bb410 (2020-08-03 13:03:31 -0400)
tilt logs: add port/resource flags, filter by resource [ch8432 ch8433]

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/hud/server/logs_reader.go [hunk 1]
```diff
@@ -67,28 +82,36 @@ func (ls *LogStreamer) Handle(v proto_webview.View) error {
...
+		}
 	}
 
-	ls.printer.Print(ls.logstore.ContinuingLines(ls.checkpoint))
+	ls.printer.Print(ls.logstore.ContinuingLines(ls.checkpointInternal))
 
-	if toCheckpoint > ls.checkpoint {
-		ls.checkpoint = toCheckpoint
+	if segsAdded > 0 {
+		ls.checkpointInternal += logstore.Checkpoint(segsAdded)
```
Comment (nicks) [inline internal/hud/server/logs_reader.go:108]: shouldn't this be `ls.logstore.Checkpoint()`, like we do in TerminalStream?

no reason to be doing checkpoint arithmetic out here

File: internal/hud/server/logs_reader.go [hunk 2]
```diff
@@ -67,28 +82,36 @@ func (ls *LogStreamer) Handle(v proto_webview.View) error {
...
+	if fromCheckpoint < ls.checkpointExternal {
 		// The server is re-sending some logs we already have, so slice them off.
-		deleteCount := ls.checkpoint - fromCheckpoint
+		deleteCount := ls.checkpointExternal - fromCheckpoint
 		segments = segments[deleteCount:]
 	}
 
-	// TODO(maia): filter for the resources that we care about (`tilt logs resourceA resourceC`)
-	//   --> and if there's only one resource, don't prefix logs with resource name?
+	// TODO(maia): if there's only one resource, don't prefix logs with resource name
```
Comment (nicks) [inline internal/hud/server/logs_reader.go:91]: i wonder if it makes more sense to have an options object to ContinuingLines that takes a set of resources to filter on....we already do this internally with toLogLines logOptions (which filters on a set of spans)

**Comment by nicks** [comment]
also - the PR description says it added port flags, but i couldn't find them?

**Comment by maiamcc** [comment]
reworked this with new code from #3664, much simpler now, ty @nicks! Nick and @jazzdan PTAL?

File: internal/hud/server/logs_reader_test.go [hunk 3]
```diff
@@ -9,11 +9,16 @@ import (
...
+
 	"github.com/tilt-dev/tilt/internal/hud"
 	proto_webview "github.com/tilt-dev/tilt/pkg/webview"
 )
 
-var defaultMessages = []string{"alpha", "bravo", "charlie", "delta"}
+var alphabet = []string{"alpha", "bravo", "charlie", "delta", "echo", "foxtrot",
+	"golf", "hotel", "igloo", "juliette", "kilo", "lima", "mike", "november",
+	"oscar", "papa", "quebec", "romeo", "sierra", "tango", "uniform", "victor",
+	"whiskey", "xavier", "yankee", "zulu"}
```
Comment (jazzdan) [inline internal/hud/server/logs_reader_test.go:21]: 🥃

Comment (maiamcc) [reply internal/hud/server/logs_reader_test.go:21]: cheers!

**Pull request merged**